src/stubs.ts

```diff
@@ -355,7 +381,7 @@ export const createFigma = (config: TConfig): PluginAPI => {
       return null;
     }
   }
-  class RectangleNodeStub {
+  class RectangleNodeStub extends BaseNodeMixinStub {
```

Comment:
Could you ask, why we need to use `extends BaseNodeMixinStub`? 

```js
applyMixins(RectangleNodeStub, [
    BaseNodeMixinStub,
    ...
]);
```    

should add BaseNodeMixinStub functionality to RectangleNodeStub (and others)

Reply:
For some reason, I've had trouble getting that to work for default properties, like `pluginData` and `sharedPluginData`. I think I was getting errors saying that those properties weren't defined

Reply:
Looking at it now, maybe it's a TypeScript issue? I'm getting errors like this:

```
FAIL  src/__tests__/create.test.ts
Test suite failed to run

    TypeScript diagnostics (customize using `[jest-config].globals.ts-jest.diagnostics` option):
    src/stubs.ts:601:16 - error TS2339: Property 'pluginData' does not exist on type 'InstanceNodeStub'.

    601       instance.pluginData = {};
                       ~~~~~~~~~~
    src/stubs.ts:602:16 - error TS2339: Property '_orig' does not exist on type 'InstanceNodeStub'.

    602       instance._orig = this;
```

Not sure what the best way to handle that would be, since it wouldn't know the type info is there until the mixins are applied 🤔

Reply:
I don't know if this is something you'd consider right now as part of this PR, but maybe a long-term goal could be to adopt something like [Constrained Mixins]

(https://www.typescriptlang.org/docs/handbook/mixins.html#constrained-mixins)

Here's an example showing a withPluginTyping mixin. I'm sure there's a way to make this more generic so we can make an `applyMixins` function, but I'm having trouble getting the types to work out  correctly 😞 

https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgAoBsCuBzUAROMOAWWAA9RkBvAKGWWwjAx30LgAoBrCATwC5kAZzBRQ2AJSCRYkNjrCmLXCAJFufaaPEAaZADc4WCFtmTB+gPbAAJgsbMsKtXADSfIRynDtcgNoAujT2TADKABZwUBA2ymzqIHAAthBCAA6IJj5mejwC2eLeMuIKQmGR0bFO8ZyJKemZprrIeU1yeobGbeYG1nYhYBFRMXGq7O68nnWpGUjdRb7YgTQAvsEwmCAIYMCWIMgA7sBg4aMuADwAKgBCcGXIEGSQIDZCyADCezKY25ZQAHwcW5lQQ3O4QCTUBT0aJgTBQfYIdB3N6PZ6vZDAlDAJJpdAQFLgN5ndikCj7Wj0KlUhBfUQ-MB-DgAOlZUWwQkEcBAvECkKoyGh1OFQkwaWgLLZUA5EgA3ILhVS1kLqQ4SeoAPqtApyfnIWHw-YAIiNyDWivoZUcrDGmu1xXayA1nUwWQdkmo5otauqtrcHi81H1TENyECZuCFoY5WGVRtLg4GumDTmOuwei1mjTeoNCOQJojUatQ0q6s4SeSM0aaYz9sWGZdbsW-K9iocJZGvpcE08FfqsybZhzIbz4dbEeVSJRyAAkiBjsAjAA5Sw2CBkyiUqkAem3mPByBXa+QG-2a5goAXX1WwVpIBE+7KR-X5EoAF5DsdTl32Bw5wvl1XF9yTlW86WQEAgOQD8QAgA5HwgZ9Ty8WVgkgtdmR9eNfyNGBLEsI0JGCIA

Reply:
An easier but hackier solution might be to do something like [this](https://www.typescriptlang.org/play?#code/PTAEEMAdIGwTwLIEsAeSB2BnUALc2AjAUyPVAFsB7AEwDMkjrQAXS0AJyOYFd2zxQ6IgHdQAY0pZm7bmNbsAUCFAZMzIuCaVaoBbW7o5SSRGjxkaLAB4AKqCIp16atiGiAFADpv4dgHNMAC4IdDgASlAAXgA+UABvAF9o92oidiQAN0YAYXlg8FCAGlACfCJcynYgkLgAbQBdMOCASSkCsSIbOEgiW1i4hVBxSTVBEQr2KNAAeQIAKyI5TzFOcHUUtMyc+TCAbkGSsonMT1pKgFFwMRx3Usxy+Sj+g6HZhaW-LmnhdAAFdkoPXYzDgADlwOQiJhbkd5J5IADWCCemFThcrjd0BCiE94i8hjN5otmJ5UvQhP9AWkQe58QSxsIJvDEZRkURCnSCVjIRz6fS3sTPJ9mN8-gCgSCACJQlZISDyGH3JkI1ms7rswTYiL4UCUiVwaWYWXyyqcvb4hLmoaW-ZDTg8PgMib7BIKJRgCwYUBkjBIZjGLDFZjgADWpFAtAB5BYOBx9D85HAAFpkRgAqAYEgCOxfHA3Rh1OxaFccb8YNw-BhJWtwJ6yAMhsKyxWqzX3GG4ME1Ol0H4mqBu2mDvdmM3K+hq8H20ROwPpGnihlwOWiF3573+xlKEhqAcm+Xx5PwABpGfQ-uD3sNN2NrgAZTwnGoY9bU+5UMgJbXPb8xQ737TC91z8Yd70fRgXwnNt30wT8OgA3s-xnBDf1AJcVxQzdt13PcwN8CCD1fE8z3cGC4NXOcfyAn9r1dBQCzSYsOlAAAhMpQRoIg63sRxSBcXVCKg4NuIGA43FAdx+zY+4ONSOt9gOVZqEkeAVGoFCDmUO0NGU9BVM-Th0GYYJ3GkohZJxAAyUBshwJAYGoQy6wiAAfQRuBgGAxOxFDQGUDj1C7Lh-V7GMkGwZdMDYEdsAAA3AbhWAAJVIbFYpYNhixge5QBMGwHGYCzMEUnSVLgDgiCoLJ1JKShKBgDR0AOVg72AySNO0qqiHatDsJvAcuBSmAEsMHAjxSGtgjiUBagkchE2cFD6mCYAACpvRldJ5QDUBVrAS8-FAS1gi3HdcOYIaRuucb+2m2bKHmgoaoO5a-PW1IjS2-0TD2yi0yOhQ6OUNRuAIbBmDwZhQGEeyYBKHE5oIDBGAyiNKmjCGcTZN0xGGzBsDMiy61a0G8XpZQzNACzQG4n10D9ANiro3H8GwSCj2JngCBUchYEq0hmDZwSOdQL0GwJfcWyE8Bp1nA6Imm+1eDIAAiFWAfpEd2bbf8-sQtDl24Cj5fiV16Ulw8a1POBz3iCqHTIBoAfxYUH3w59heg7FYK-PXUN1k3Fa4ZXQDVjWCRHN2n21t9vfIlCkLl4DF0N43gIVs2Jbw6PPeDa3oTI33A-tkOnddZmRihwnOO4yJTFgRBResau5Kb5IW64puSYIYpahj2su65xp9gkKRBE4qZxI7utJIU9BOKFLh+-cFWzkoFWwgUIA)

This would modify `applyMixins` to return a new class rather than modifying an existing one, and it accepts a generic argument that specifies the output type. There's really nothing in the function to enforce that, since `baseCtor` and `derivedCtors` are `any` types, but that's already the case